js/script.js

```diff
@@ -0,0 +1,481 @@
+
+            tempObj = this.setValue(path,this.getValue(fromPath,tempObj),tempObj);      
+            this.updateHTMLOutput(tempObj);                  
+
+            if(isNaN(key)){
+              matchText = JSON.stringify(fromKey) + ': ' +JSON.stringify(this.getValue(fromPath,this.jsonObj)); 
+            }
+            else{ 
+              matchText = JSON.stringify(this.getValue(fromPath,this.jsonObj)); 
+            }
+
+            console.log(matchText);
+
+            this.jsonHTML = this.jsonHTML.replace(
+              matchText,'<span class="old">' + matchText + '</span>'
+            );
+            break;
+          case 'copy':      
+            fromKey = fromPath[fromPath.length - 1];   
+            if(isNaN(fromKey)){ 
+              matchText = JSON.stringify(fromKey) + ': ' +JSON.stringify(this.getValue(fromPath,this.newJsonObj)); 
+            }
+            else{ 
+              matchText = JSON.stringify(this.getValue(fromPath,this.newJsonObj)); 
+            }          
+
+            this.updateHTMLOutput(this.newJsonObj);                   
+            this.jsonHTML = this.jsonHTML.replace(
+              matchText,'<span class="new">' + matchText + '</span>'
+            );
+            break;
+          case 'test':
+            break;
+        }
+      }
+      else{
+        alert('You should execute the patches in order');
+      }
+    },
+
+    getPath(path){ // Get the JSON patch string from the array of Path. eg [a,b,c,1] should return a.b.c[1];
+      let pathString = path[0];
+      for(let i=1;i< path.length;i++){
+        if(isNaN(path[i])){
+          pathString += '.' + path[i];
+        }
+        else{
+          pathString += '[' + path[i] + ']';
+        }
+      }
+      return pathString;
+    },    
+
+    // Method to add proprerty to the Object 
+    setValue(path, value, newObj){  
+      if(!newObj) newObj = this.newJsonObj; 
+
+      let exec = 'newObj.' + this.getPath(path) + 
+                  '=' + 
+                  (typeof(value) === 'object' ? ('JSON.parse(\'' + JSON.stringify(value) + '\')') : ('\'' + value + '\''));                           
+      eval(exec); 
+      // The Eval function will execute the string command eg. if exec = 'delete Object.value', if will execute the statement.            
+      return newObj;
+    },
+
+    // Method to delete propery from the object.
+    unsetValue: function(path, newObj = null){ 
+      if(!newObj) newObj = this.newJsonObj;
+      let exec = 'delete newObj.' + this.getPath(path);      
+      eval(exec);       
+      // The Eval function will execute the string command eg. if exec = 'delete Object.value', if will execute the statement.                 
+      return newObj;      
+    },
+
+    // Method to get value from a path and object that's passed.
+    getValue(path,obj = null){ 
+      if(!obj) obj = this.newJsonObj;     
+      let exec = 'obj.' + this.getPath(path);
+      return eval(exec);
```

Comment:
Can we access nested properties without using the unsafe eval ?

Reply:
Yes, I will try finding the path using recursion rather than using eval expression.

Reply:
Yes, I will try finding the path using recursion rather than using eval expression.